indico/modules/events/sessions/controllers/display.py

```diff
@@ -79,7 +86,45 @@ def _process(self):
                          'text/calendar')
 
 
-class RHExportSessionTimetableToPDF(RHDisplaySessionBase):
+class RHExportSessionTimetableToPDF(RHDisplaySessionBase, RHTimetableExportPDF):
     def _process(self):
-        pdf = get_session_timetable_pdf(self.session)
-        return send_file('session-timetable.pdf', BytesIO(pdf.getPDFBin()), 'application/pdf')
+        now = now_utc()
+        css = render_template('events/timetable/pdf/timetable.css')
+        event = self.event
+        entries = [
+            e for e in get_nested_timetable(event)
+            if e.type == TimetableEntryType.SESSION_BLOCK and (
+                e.id == self.session.id or (e.session_block and self.session.id == e.session_block.session.id)
+            )
+        ]
+        days = {
+            day: list(e) for day, e in groupby(
+                entries, lambda e: e.start_dt.astimezone(self.event.tzinfo).date()
+            )
+        }
+        config = TimetableExportConfig(
+            show_title=True,
+            show_affiliation=False,
+            show_cover_page=False,
+            show_toc=False,
+            show_session_toc=True,
+            show_abstract=False,
+            dont_show_poster_abstract=False,
+            show_contribs=False,
+            show_length_contribs=False,
+            show_breaks=False,
+            new_page_per_session=False,
+            show_session_description=False,
+            print_date_close_to_sessions=False
+        )
+
+        show_siblings_location, show_children_location = get_nested_timetable_location_conditions(entries)
+        program_config = TimetableExportProgramConfig(
+            show_siblings_location=show_siblings_location,
+            show_children_location=show_children_location
+        )
```

Comment:
Since we don't have a parent (event) location in this particular view, I think we always want to show the sibling locations here - otherwise you are likely to not have any location information in the PDF at all

```suggestion
        show_children_location = get_nested_timetable_location_conditions(entries)[1]
        program_config = TimetableExportProgramConfig(
            show_siblings_location=True,
            show_children_location=show_children_location
        )
```